common/src/main/scala/org/apache/spark/sql/comet/execution/arrow/ArrowWriters.scala

```diff
@@ -137,6 +150,28 @@ private[arrow] abstract class ArrowFieldWriter {
     count += 1
   }
 
+  def writeCol(input: ColumnarArray): Unit = {
+    val inputNumElements = input.numElements()
+    valueVector.setInitialCapacity(inputNumElements)
+    while (count < inputNumElements) {
+      if (input.isNullAt(count)) {
+        setNull()
+      } else {
+        setValue(input, count)
+      }
```

Comment:
This comment isn't related to any changes in this PR, but I was a little surprised at our current api where we call `setNull()` with no arguments and it knows to set null at the current index, but when we call `setValue` we pass the index. It just seems inconsistent. We can maybe revisit this design in the future.

Reply:
Cruft from the original implementation from Spark's `ArrowWriter`, it seems.